test_runner/src/main/kotlin/ftl/reports/util/ReportManager.kt

```diff
@@ -40,25 +39,11 @@ object ReportManager {
         return xmlFiles
     }
 
-    private fun getWebLink(matrices: MatrixMap, xmlFile: File): String {
-        // xmlFile path changes based on if local-result-dir is used. may or may not contain objName
-        // 2019-03-22_17-20-53.594000_ftrh/shard_0/test_result_1.xml or shard_0/test_result_1.xml
-        val objName = matrices.runPath // 2019-03-22_17-20-53.594000_ftrh
-
-        // shard location in path changes based on iOS or Android.
-        val matchResult = Regex("/.*(shard_\\d+)(-rerun_\\d+)?/").find(xmlFile.toString())
-        val shardName = matchResult?.value?.removePrefix("/")?.removeSuffix("/") // shard_0 || shard_0-rerun_1
-        val matrixPath = Paths.get(objName, shardName).toString() // 2019-03-22_17-20-53.594000_ftrh/shard_0
-
-        var webLink = ""
-        val savedMatrix = matrices.map.values.firstOrNull { it.gcsPath.endsWith(matrixPath) }
-        if (savedMatrix != null) {
-            webLink = savedMatrix.webLink
-        } else {
-            println("WARNING: Matrix path not found in JSON. $matrixPath")
-        }
-        return webLink
-    }
+    private fun getWebLink(matrices: MatrixMap, xmlFile: File): String = xmlFile.getMatrixPath(matrices.runPath)
+        ?.let { path ->
+            matrices.map.values.firstOrNull { savedMatrix -> savedMatrix.gcsPath.endsWith(path) }?.webLink
+                ?: "".also { println("WARNING: Matrix path not found in JSON. $path") }
```

Comment:
Could you extract this to separate method for better readability?

Reply:
Changed!